base/c.jl

```diff
@@ -23,7 +23,7 @@ if is_windows()
 else
     typealias Clong Int
     typealias Culong UInt
-    typealias Cwchar_t Int32
+    typealias Cwchar_t UInt32
```

Comment:
I know that Cwchar_t is passed as argument to the function `transcode` and it seems to accept on all the cases unsigned integers. So I changed the alias from `Int32` to `UInt32`. Also now it is coherent with the definition of the aliases for windows.

On the definition of `transcode{T::Union{UInt8, UInt16}(::Type{T}, src::T)=src`, line 143 and beyond.

I'm trying to build with appveyor but it is still running. I was able to build it locally with no error